frontend/src/fusion.ts

```diff
@@ -0,0 +1,94 @@
+// SocketManager wraps a WebSocket in order to provide guarantees about
+// reliability, reconnections, retries, etc.
+class SocketManager {
+    ws: WebSocket
+    url: string
+
+    constructor(url: string) {
+        this.url = url;
+        this.ws = this.createSocket();
+    }
+
+    createSocket() {
+        const ws = new WebSocket(this.url);
+        ws.onopen = event => {
+            // console.log("socket connected", event);
+        };
+        ws.onmessage = event => {
+            // console.log(event);
+        }
+        ws.onerror = event => {
+            // console.log("socket error", event);
+        };
+        ws.onclose = event => {
+            // console.log("socket closed", event);
+            this.ws = this.createSocket();
+        };
+        return ws;
+    }
+
+    send(msg: string) {
+        this.ws.send(msg);
+    }
+}
+
+export default class Fusion {
```

Comment:
Current linter rules don't allow multiple classes per file, either change this or add `max-classes-per-file: false` to tsconfig.json